### PR TITLE
Updated many VisualServer functions

### PR DESCRIPTION
**Pull request by clayjohn**
Another big batch of VisualServer function documentation. 

I'm trying to keep these in easy-to-review chunks. Let me know If it is best to just put it all in one PR and I can just force push the new changes as I make them.

### CODE REVIEW

File: doc/classes/VisualServer.xml [hunk 1]
```diff
@@ -1397,6 +1406,9 @@
 			<return type="RID">
 			</return>
 			<description>
+				Creates a gi probe and adds it to the VisualServer. It can be accessed with the RID that is returned. This RID will be used in all [code]gi_probe_*[/code] VisualServer functions.
```
Comment (akien-mga) [inline doc/classes/VisualServer.xml]: "gi" should be capitalized ("GI probe"), as it's an acronym (well, initialism, but let's not be pedantic :P).

File: doc/classes/VisualServer.xml [hunk 2]
```diff
@@ -1610,6 +1622,7 @@
 			<argument index="0" name="feature" type="int" enum="VisualServer.Features">
 			</argument>
 			<description>
+				Not yet implemented. Always returns false.
```
Comment (akien-mga) [inline doc/classes/VisualServer.xml]: Should be `[code]false[/code]`.

File: doc/classes/VisualServer.xml [hunk 3]
```diff
@@ -1753,13 +1768,14 @@
...
 			<description>
+				Adds the next vertex using the information provided in advance. This is a helper class that calls [method immediate_vertex] under the hood. Equivalent to [method ImmediateGeometry.add_vertex].
 			</description>
 		</method>
 		<method name="init">
 			<return type="void">
 			</return>
 			<description>
-				Initializes the visual server.
+				Initializes the visual server. This function is called internally by platform dependent code during engine initialization. If called from a running game, it will not do anything.
```
Comment (akien-mga) [inline doc/classes/VisualServer.xml]: There should be a hyphen between "platform" and "dependent" IINM.

File: doc/classes/VisualServer.xml [hunk 4]
```diff
@@ -1770,6 +1786,7 @@
 			<argument index="1" name="id" type="int">
 			</argument>
 			<description>
+				Attaches unique Object ID to instance. Object ID must be attached to instance for proper culling with [method instances_cull_aabb], [method instances_cull_convex], and [method instances_cull_ray].
```
Comment (akien-mga) [inline doc/classes/VisualServer.xml]: "**a** unique object ID"

File: doc/classes/VisualServer.xml [hunk 5]
```diff
@@ -1864,6 +1892,7 @@
 			<argument index="1" name="base" type="RID">
 			</argument>
 			<description>
+				Sets the base of the instance. A base can be any of the 3D objects that are created in the VisualServer that can be displayed. For example, any of the light types, mesh, multimesh, immediate geometry, particles, reflection probe, lightmap capture, and the gi probe are all types that can be set as the base of an instance in order to be displayed in the scenario.
```
Comment (akien-mga) [inline doc/classes/VisualServer.xml]: **GI** probe

File: doc/classes/VisualServer.xml [hunk 6]
```diff
@@ -2817,12 +2867,18 @@
...
+				Creates a new omni light and adds it to the VisualServer. It can be accessed with the RID that is returned. This RID can be used in most [code]light_*[/code] VisualServer functions.
+				Once finished with your RID, you will want to free the RID using the VisualServer's [method free_rid] static method.
+				To place in a scene, attach this omni light to an instance using [method instance_set_base] using the returned RID.
 			</description>
 		</method>
 		<method name="particles_create">
 			<return type="RID">
 			</return>
 			<description>
+				Creates a particles and adds it to the VisualServer. It can be accessed with the RID that is returned. This RID will be used in all [code]particles_*[/code] VisualServer functions.
```
Comment (akien-mga) [inline doc/classes/VisualServer.xml]: "a particles" sounds wrong. Could be "a particles subsystem" maybe.

File: doc/classes/VisualServer.xml [hunk 7]
```diff
@@ -3410,6 +3476,9 @@
 			<return type="RID">
 			</return>
 			<description>
+				Creates a spot_light and adds it to the VisualServer. It can be accessed with the RID that is returned. This RID can be used in most [code]light_*[/code] VisualServer functions.
```
Comment (akien-mga) [inline doc/classes/VisualServer.xml]: You kept the underscore in "spot_light".

**Comment by clayjohn** [comment]
@akien-mga Thanks for the feedback!

**Comment by akien-mga** [comment]
Thanks a lot!

**Pull request merged**